test/tools/router_check/router.cc

```diff
@@ -183,13 +183,11 @@ bool RouterCheckTool::compareEntries(const std::string& expected_routes) {
     tool_config.route_ = config_->route(*tool_config.headers_, tool_config.random_value_);
 
     const std::string& test_name = check_config.test_name();
-    if (details_) {
-      std::cout << test_name << std::endl;
-    }
+    tests_.insert({test_name, {}});
```

Comment:
LGTM, the printing step is in one place now and feels concise to me.
However, using a map will cause issues while printing because the test names are not guarenteed to be unique. So all tests with the same name will be clubbed together. We solve that using something that is not a map.

Reply:
Used a vector of pairs of strings and string vectors (so of the form <(testname1, test messages), (testname2, test messages)...>).

Reply:
Replaced map with a vector of pairs